src/features/Chat/components/ActiveWindow/index.tsx

```diff
@@ -0,0 +1,234 @@
+import { useState } from 'react';
+
+import {
+  ChatFolder,
+  useSendMessageMutation,
+  toSendMessage,
+  useUpdateStatusMutation,
+} from '@/features/Chat/chatPageApi';
+import { useAppSelector } from '@/store';
+import {
+  selectActiveChat,
+  selectIsLoadingBuddyMessages,
+} from '@/features/Chat/chatSlice';
+import { selectUserId } from '@/features/Authentication/userSlice';
+
+import { useTranslation } from 'react-i18next';
+
+import styled from 'styled-components';
+import { palette } from '@/components/variables';
+import { Profile as ProfileIcon } from '@/components/Icons/Profile';
+import ArchivedIcon from '@/static/icons/archived-chats.svg';
+import BlockedIcon from '@/static/icons/blocked-chats.svg';
+import Text from '@/components/Text';
+import TextInput from '@/components/TextInput';
+import { IconButton, StatusButton } from '@/components/Buttons';
+import { MessageList } from './MessageList';
+
+const searchInputIconSize = 24;
+const closeInputIconSize = 34;
+
+const ActiveWindow = () => {
+  const { t } = useTranslation('chat');
+  const [showSearch, setShowSearch] = useState(false);
+  const [searchValue, setSearchValue] = useState('');
+  const [text, setText] = useState('');
+  const [sendMessage, { isLoading: isMessageSendLoading }] =
+    useSendMessageMutation();
+  const [updateChatStatus] = useUpdateStatusMutation();
+  const userId = useAppSelector(selectUserId);
+
+  const chat = useAppSelector(selectActiveChat);
+  const isLoadingMessages = useAppSelector(
+    selectIsLoadingBuddyMessages(chat?.buddyId),
+  );
+
+  const updateStatus = (status: ChatFolder) => {
+    if (!userId || !chat) return;
+    updateChatStatus({ userId, buddyId: chat.buddyId, status });
+  };
+
+  const handleMessageSend = (buddyId: string, text: string) => {
+    if (!userId || isMessageSendLoading) return;
+
+    const message = toSendMessage(buddyId, userId, text);
+    sendMessage({ userId, message });
+    setText('');
+  };
+
+  const isLoading = isLoadingMessages || isMessageSendLoading;
+
+  const getProfileIcon = (status: ChatFolder) => {
+    const icons = {
+      ok: <ProfileIcon color="purpleDark" />,
+      archived: <img src={ArchivedIcon} />,
+      banned: <img src={BlockedIcon} />,
+    };
+
+    return icons[status];
+  };
+
+  return (
+    chat && (
+      <ActiveChatContainer>
+        <HeaderBar>
+          <ProfileInfo>
+            {getProfileIcon(chat.status)}
```

Comment:
This can also be just a map, no need to make a function
It can then be used just with 

```
icons[chatStatus]
```

Reply:
Great point! I'll change it right away 👍